src/meta/processors/job/JobManager.cpp

```diff
@@ -83,14 +83,26 @@ nebula::cpp2::ErrorCode JobManager::handleRemainingJobs() {
       auto optJob = nebula::value(optJobRet);
       std::unique_ptr<JobExecutor> je =
           JobExecutorFactory::createJobExecutor(optJob, kvStore_, adminClient_);
-      // Only balance has been recovered
+      // Only balance would change
       if (optJob.getStatus() == cpp2::JobStatus::RUNNING && je->isMetaJob()) {
-        jds.emplace_back(optJob);
+        jds.emplace_back(std::move(optJob));
```

Comment:
Won't this would break the rule below?
```
for a balance job:
(1) if there's a newer finished balance job, the stopped or failed balance job can't be recovered.
(2) only the lasted stopped or failed balance job could be recovered
```

Reply:
the guarantee is at jobmanager.cpp:832